app/lib/package/backend.dart

```diff
@@ -504,52 +501,35 @@ class PackageBackend {
       package.updateVersion(newVersion);
       package.updated = DateTime.now().toUtc();
 
-      try {
-        _logger.info(
-          'Trying to upload tarball for ${package.name} version ${newVersion.version} to cloud storage.',
-        );
-        // Apply update: Push to cloud storage
-        await tarballUpload(package.name, newVersion.version);
-
-        final inserts = <Model>[
-          package,
-          newVersion,
-          validatedUpload.packageVersionPubspec,
-          validatedUpload.packageVersionInfo,
-          ...validatedUpload.assets,
-        ];
-        if (historyBackend.isEnabled) {
-          final history = History.entry(PackageUploaded(
-            packageName: newVersion.package,
-            packageVersion: newVersion.version,
-            uploaderId: user.userId,
-            uploaderEmail: user.email,
-            timestamp: newVersion.created,
-          ));
-          inserts.add(history);
-        }
-
-        // Apply update: Update datastore.
-        _logger.info('Trying to commit datastore changes.');
-        T.queueMutations(inserts: inserts);
-        await T.commit();
-
-        _logger.info('Upload successful. [package-uploaded]');
-        return newVersion;
-      } catch (error, stack) {
-        _logger.warning('Error while committing: $error, $stack');
-
-        // This call might fail if the transaction has already been
-        // committed/rolled back or the transaction failed.
-        //
-        // In which case we simply ignore the rollback error and rethrow the
-        // original error.
-        try {
-          await T.rollback();
-        } catch (_) {}
-        rethrow;
+      _logger.info(
+        'Trying to upload tarball for ${package.name} version ${newVersion.version} to cloud storage.',
+      );
+      // Apply update: Push to cloud storage
+      await tarballUpload(package.name, newVersion.version);
+
+      final inserts = <Model>[
+        package,
+        newVersion,
+        validatedUpload.packageVersionPubspec,
+        validatedUpload.packageVersionInfo,
+        ...validatedUpload.assets,
+      ];
+      if (historyBackend.isEnabled) {
+        final history = History.entry(PackageUploaded(
+          packageName: newVersion.package,
+          packageVersion: newVersion.version,
+          uploaderId: user.userId,
+          uploaderEmail: user.email,
+          timestamp: newVersion.created,
+        ));
+        inserts.add(history);
       }
+
+      _logger.info('Trying to commit datastore changes.');
+      tx.queueMutations(inserts: inserts);
```

Comment:
nit... just use `tx.insert` then we don't need to make a list of `inserts` at all :)

Reply:
I've moved the history entry inside the `inserts` list. It doesn't seem to worth switching to individual `tx.insert` calls with that...